junit5/src/main/java/cz/xtf/junit5/listeners/OpenshiftStateRecorder.java

```diff
@@ -0,0 +1,199 @@
+package cz.xtf.junit5.listeners;
+
+import cz.xtf.core.config.BuildManagerConfig;
+import cz.xtf.core.config.XTFConfig;
+import cz.xtf.core.openshift.OpenShift;
+import cz.xtf.core.openshift.OpenShifts;
+import io.fabric8.kubernetes.api.model.ContainerStatus;
+import io.fabric8.kubernetes.api.model.Event;
+import io.fabric8.kubernetes.api.model.Pod;
+import io.fabric8.kubernetes.api.model.Secret;
+import io.fabric8.kubernetes.api.model.Service;
+import io.fabric8.kubernetes.client.KubernetesClientException;
+import io.fabric8.openshift.api.model.DeploymentConfig;
+import io.fabric8.openshift.api.model.Route;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.LifecycleMethodExecutionExceptionHandler;
+import org.junit.platform.engine.TestExecutionResult;
+import org.junit.platform.launcher.TestExecutionListener;
+import org.junit.platform.launcher.TestIdentifier;
+
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.LocalDateTime;
+
+import static org.junit.platform.engine.TestExecutionResult.Status.FAILED;
+
+/**
+ * Listener for storing information from openshift in case a test fails.
+ * It stores pod logs, openshift routes, secret, events etc. From both work and build namespace.
+ * It implements two distinct interfaces TestExecutionListener and LifecycleMethodExecutionExceptionHandler
+ * both serving different purpose and need to be applied differently.
+ *
+ * TestExecutionListener
+ *    Is called whenever test ends. Stores openshift state if test failed.
+ *    To use this interface add this class as a listener
+ *
+ * LifecycleMethodExecutionExceptionHandler
+ *    This interface is executed when @BeforeAll or @BeforeEach method fails - throw an exception.
+ *    To use this interface a test class needs to be extended with this class using @ExtendWith
+ */
+@Slf4j
+public class OpenshiftStateRecorder implements TestExecutionListener, LifecycleMethodExecutionExceptionHandler {
+		private static final Path statusDir = Paths.get("log","status");
```

Comment:
Why two tabs for space? All the project uses one tab everywhere...

Reply:
@Crumby Fixed.